// node-bot/lib/price.js
var request = require('request'),
	dom = require('node-dom').dom,
	fs = require('fs'),   
	URL = require('url');


var getUrl=function(query,Response) {
	
	var params={}; //url-name

	var $E=encodeURIComponent;

	query.split('&').forEach(function(param){
		var parts = param.split('=');
		params[parts[0].trim()] = (decodeURIComponent(parts[1]) || '').trim();
	});
	
	var req = {uri:params.url};
	
	console.log(req);

	request(req,function (error, response, page) {
			
			var options =	{	url:params.url,
								features: {
											//FetchExternalResources  : {script:'', img:'', input:'', link:''},
											//ProcessExternalResources: {script:'',img:'',link:'',input:''},
											FetchExternalResources  : {script:''},
											ProcessExternalResources: {script:''},
											removeScript: true //Remove scripts for innerHTML and outerHTML output
								},
								forms:true
			};
			
			window=dom(page,null,options);
			
			document=window.document;
			
			document.onload=function() {
			
				var head = {'Content-Type': 'text/javascript' };
							
				if (Response.writeHead) {
				
					Response.writeHead(200, head);
					
					console.log(params.name+'='+JSON.stringify(this._options.forms)+';');
					
					Response.end(params.name+'='+JSON.stringify(this._options.forms)+';');
				
				}
			};
	});

};

exports.getUrl=getUrl;
